target/cheri-common/cheri_defs.h

```diff
@@ -38,90 +38,70 @@
-#  define CAP_UPERMS_SHFT CC128_UPERMS_SHFT
-#  define CAP_MAX_UPERM CC128_MAX_UPERM
-#  define CAP_MAX_REPRESENTABLE_OTYPE CC128_MAX_REPRESENTABLE_OTYPE
-#  define CAP_LAST_NONRESERVED_OTYPE CC128_LAST_NONRESERVED_OTYPE
-#  define CAP_OTYPE_UNSEALED CC128_OTYPE_UNSEALED
-#  define CAP_OTYPE_UNSEALED_SIGNED CC128_OTYPE_UNSEALED_SIGNED
-#  define CAP_OTYPE_SENTRY CC128_OTYPE_SENTRY
-#  define CAP_FIRST_SPECIAL_OTYPE_SIGNED CC128_FIRST_SPECIAL_OTYPE_SIGNED
-#  define CAP_LAST_SPECIAL_OTYPE_SIGNED CC128_LAST_SPECIAL_OTYPE_SIGNED
-#  define CAP_FLAGS_ALL_BITS CC128_FIELD_FLAGS_MASK_NOT_SHIFTED
-#  define CHERI_CAP_SIZE  16
-#  ifdef TARGET_WORDS_BIGENDIAN
-#    define CHERI_MEM_OFFSET_CURSOR 8
-#    define CHERI_MEM_OFFSET_METADATA 0
-#  else
-#    define CHERI_MEM_OFFSET_CURSOR 0
-#    define CHERI_MEM_OFFSET_METADATA 8
-#  endif
-#elif defined(CHERI_256)
-#  define CAP_PERMS_ALL CC256_PERMS_ALL_BITS
-#  define CAP_UPERMS_ALL CC256_UPERMS_ALL_BITS
-#  define CAP_UPERMS_SHFT CC256_UPERMS_SHFT
-#  define CAP_MAX_UPERM CC256_MAX_UPERM
-#  define CAP_MAX_REPRESENTABLE_OTYPE CC256_MAX_REPRESENTABLE_OTYPE
-#  define CAP_LAST_NONRESERVED_OTYPE CC256_LAST_NONRESERVED_OTYPE
-#  define CAP_OTYPE_UNSEALED CC256_OTYPE_UNSEALED
-#  define CAP_OTYPE_SENTRY CC256_OTYPE_SENTRY
-#  define CAP_FIRST_SPECIAL_OTYPE_SIGNED CC256_FIRST_SPECIAL_OTYPE_SIGNED
-#  define CAP_LAST_SPECIAL_OTYPE_SIGNED CC256_LAST_SPECIAL_OTYPE_SIGNED
-#  define CAP_FLAGS_ALL_BITS CC256_FLAGS_ALL_BITS
-#  define CHERI_CAP_SIZE  32
-#  ifdef TARGET_WORDS_BIGENDIAN
-#    define CHERI_MEM_OFFSET_METADATA 0
-#    define CHERI_MEM_OFFSET_CURSOR 8
-#    define CHERI_MEM_OFFSET_BASE 16
-#    define CHERI_MEM_OFFSET_LENGTH 24
-#  else
-#    error no little endian CHERI256
-#  endif
+
+#if TARGET_LONG_BITS == 32
+#  define CHERI_CAP_SIZE 8
+#  define CHERI_CAP_BITS 64
+#elif TARGET_LONG_BITS == 64
+#  define CHERI_CAP_SIZE 16
+#  define CHERI_CAP_BITS 128
 #else
-#  error "Unsupported capability type"
+#  error "Unknown CHERI capability format"
 #endif
 
-/* compressed capabilities use a 65-bit top, precise and magic 128 64-bits */
-#if defined(CHERI_128)
-#  define CAP_MAX_LENGTH CC128_MAX_LENGTH
-#  define CAP_MAX_TOP CC128_MAX_TOP
-#elif defined(CHERI_256) || defined(CHERI_MAGIC128)
-#  define CAP_MAX_LENGTH CC256_NULL_LENGTH
-#  define CAP_MAX_TOP CC256_NULL_TOP
+#ifdef TARGET_WORDS_BIGENDIAN
+#  define CHERI_MEM_OFFSET_CURSOR TARGET_LONG_SIZE
+#  define CHERI_MEM_OFFSET_METADATA 0
 #else
-#  error "Unsupported capability type"
+#  define CHERI_MEM_OFFSET_CURSOR 0
+#  define CHERI_MEM_OFFSET_METADATA TARGET_LONG_SIZE
 #endif
 
-typedef cc128_cap_t cap_register_t;
-typedef signed __int128 cap_offset_t;
-typedef unsigned __int128 cap_length_t;
+/*
+ * Optional suffix intended for Morello to differentiate its 128-bit
+ * compression format from the CHERI-MIPS and CHERI-RISC-V CHERI-128 format,
+ * allowing e.g. cc128 and cc128m to coexist. Unused for now.
+ */
+#define CHERI_CAP_VARIANT_LOWER
```

Comment:
I wonder if this should be moved to cheri-compressed-cap so that e.g. GDB can use 64+128 more easily.